Sources/MapboxMapsSwiftUI/MapCoordinator.swift

```diff
@@ -96,3 +106,27 @@ public final class MapCoordinator {
         }.addTo(queriesBag)
     }
 }
+
+private func wrapError(_ body: () throws -> Void) {
+    do {
+        try body()
+    } catch {
+        print("error: \(error)") // TODO: Logger
+    }
+}
+
+private func set<T: Equatable>(_ getter: () -> T, _ setter: (T) throws -> Void, value: T) {
+    wrapError {
+        if getter() != value {
+            try setter(value)
+        }
+    }
+}
+
+private func set<T: Equatable>(_ oldValue: T, _ setter: (T) throws -> Void, value: T) {
+    set({ oldValue }, setter, value: value)
+}
+
+private func set<U: AnyObject, T: Equatable>(_ object: U, _ keyPath: ReferenceWritableKeyPath<U, T>, value: T) {
+    set({ object[keyPath: keyPath] }, { object[keyPath: keyPath] = $0 }, value: value)
+}
```

Comment:
Should we extract global level function to some context? At least we have to have a separate file for them in my understanding.

Reply:
Yep, I was thinking about that, yet found out that they are actually quite tied to what we need here in MapCoordinator. At least, the error handled in certain non-generic way and properties can be set in several non-standardized ways.